lib/dangermattic/plugins/manifest_pr_checker.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check if the Gemfile.lock was updated when changing the Gemfile in a PR.
+  class ManifestPRChecker < Plugin
+    # Performs all the checks, asserting that changes on `Gemfile`, `Podfile` and `Package.swift` must have corresponding
+    # lock file changes.
+    def check_all_manifest_lock_updated
+      check_gemfile_lock_updated
+      check_podfile_lock_updated
+      check_swift_package_resolved_updated
+    end
+
+    # Check if the `Gemfile` file was modified without a corresponding `Gemfile.lock` update
+    def check_gemfile_lock_updated
+      check_manifest_lock_updated(
+        file_name: 'Gemfile',
+        lock_file_name: 'Gemfile.lock',
+        instruction: 'Please run `bundle install` or `bundle update <updated_gem>`'
+      )
+    end
+
+    # Check if the `Podfile` file was modified without a corresponding `Podfile.lock` update
+    def check_podfile_lock_updated
+      check_manifest_lock_updated(
+        file_name: 'Podfile',
+        lock_file_name: 'Podfile.lock',
+        instruction: 'Please run `bundle exec pod install`'
+      )
+    end
+
+    # Check if the `Package.swift` file was modified without a corresponding `Package.resolved` update
+    def check_swift_package_resolved_updated
+      check_manifest_lock_updated(
+        file_name: 'Package.swift',
+        lock_file_name: 'Package.resolved',
+        instruction: 'Please resolve the Swift packages in Xcode'
+      )
+    end
+
+    private
+
+    def check_manifest_lock_updated(file_name:, lock_file_name:, instruction:)
+      manifest_modified_file = git.modified_files.find { |f| f.end_with?(file_name) }
+      lock_modified = git.modified_files.any? { |f| f.end_with?(lock_file_name) && manifest_modified_file && File.dirname(f) == File.dirname(manifest_modified_file) }
```

Comment:
Same limitation of this only accounting for the case of only having a single occurrence of the manifest+lockfile pair in the whole repo, instead of accounting for the possibility of having multiple `Podfile` & `Podfile.lock` in the same repo (like Tumblr-iOS does)

Reply:
Maybe something like this
```ruby
# Find all the modified manifest files
manifest_modified_files = git.modified_files.select { |f| File.basename(f) == file_name }
# For each one, check if the corresponding lockfile (in the same dir) was also modified
manifest_modified_files.each do |manifest_file|
  lockfile_modified = git.modified_files.include? do |f|
    File.dirname(f) == File.dirname(manifest_file) && File.basename(f) == lock_file_name
  end
  next if lockfile_modified

  warn("#{manifest_file} was changed without updating its corresponding #{lock_file_name}. #{instruction}."
end
```

PS: This is just a suggestion, not tested. In particular, we need to make sure (via additional unit tests ideally to cover those cases too) that my `File.join(File.dirname(manifest), lock_file_name)` will still work even if `git.modified_files` contains

Reply:
Ah, I got attached to the fact of "custom located Podfile" but of course multiple ones can exist 🙂
Good suggestion, gonna try something similar.